src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -1394,17 +1394,23 @@ mongoc_uri_finalize_tls (mongoc_uri_t *uri, bson_error_t *error)
       return false;
    }
 
+#ifndef MONGOC_ENABLE_SSL
+   if (mongoc_uri_get_tls (uri)) {
+      MONGOC_URI_ERROR (error, "SSL is not enabled in this build");
+      return false;
+   }
+#endif
```

Comment:
Is there any concern moving this check from `mongoc_client_pool_new` and `_mongoc_client_new_from_uri` to `mongoc_uri_finalize_tls` (called via `mongoc_uri_finalize`)? It did require changing a test that previously expected `mongoc_client_pool_new` to return `NULL` (that test wasn't asserting a logged error but probably should have).

This also feels more consistent because now the error message is relayed via `bson_error_t` instead of being logged directly via the `MONGOC_ERROR` macro.

Reply:
This seems to introduce some test failures, so I'm going to revert the change. One particular failure is in `test_srv_polling_mocked`, which expects to construct a `mongoc_uri_t` but never a client or pool. There's really no reason that test should require SSL to be enabled.

Reply:
> There's really no reason that test should require SSL to be enabled.

Agreed. Removing `?tls=true` from the test's URI still passes locally for me.